lib/IRGen/MetadataRequest.cpp

```diff
@@ -3373,6 +3388,7 @@ MetadataResponse
 IRGenFunction::emitTypeMetadataRef(CanType type,
                                    DynamicMetadataRequest request) {
   if (type->getASTContext().LangOpts.hasFeature(Feature::Embedded) &&
+      !type->getASTContext().LangOpts.hasFeature(Feature::EmbeddedExistentials) &&
```

Comment:
It's better to pull the ASTContext out only once into a local variable. Also there's a third usage below you could simplify